optimum/habana/transformers/trainer.py

```diff
@@ -1358,6 +1358,62 @@ def _maybe_log_save_evaluate(
             timer.step()
             self.log_evaluate_save_time += timer.last_duration
 
+    def _save_checkpoint(self, model, trial):
+        # Copied from https://github.com/huggingface/transformers/blob/v4.51-release/src/transformers/trainer.py#L3187
+        # In all cases, including ddp/dp/deepspeed, self.model is always a reference to the model we
+        # want to save except FullyShardedDDP.
+        # assert unwrap_model(model) is self.model, "internal model should be a reference to self.model
+
+        # Save model checkpoint
+        checkpoint_folder = f"{PREFIX_CHECKPOINT_DIR}-{self.state.global_step}"
+
+        if self.hp_search_backend is None and trial is None:
+            self.store_flos()
+
+        run_dir = self._get_output_dir(trial=trial)
+        output_dir = os.path.join(run_dir, checkpoint_folder)
+        self.save_model(output_dir, _internal_call=True)
+
+        if self.args.parallel_mode == ParallelMode.DISTRIBUTED:
```

Comment:
Let's also add a comment above this line to quickly explain why we add it

Reply:
Done